hack/make-rules/test.sh

```diff
@@ -261,12 +266,12 @@ runTests() {
         \"\${_pkg}\" \
         ${testargs[*]:+${testargs[*]}} \
       | tee ${junit_filename_prefix:+\"${junit_filename_prefix}-\$_pkg_out.stdout\"} \
-      | grep \"${go_test_grep_pattern}\"" \
+      " \
     {} \
+    2> "${KUBE_TEMP}/stderr-pipe" \
+    | produceJUnitXMLReport "${junit_filename_prefix}" "${KUBE_TEMP}/stderr-pipe" \
```

Comment:
Can we depend on mkfifo? It is used to pass also stderr into gotestsum, something that wasn't done before. We do not absolutely need it, but doing so makes the JUnit a bit more complete.

The alternative is to not use a pipe at all. Then gotestsum can also capture a non-zero exit code, we don't need mkfifo, and https://github.com/gotestyourself/gotestsum/pull/415 doesn't need to be merged.

To make this work, the current `printf | bash -c` magic would have to be done inside a script that gets invoked by `gotestsum --raw-command`. My first commit does that, but it's most likely not handling quoting correctly. I'm not sure whether that can be fixed. Perhaps by having a separate helper script that is passed additional parameters?

/assign @dims

Reply:
is it possible to behave somewhat ok even if `mkfifo` is missing then it's ok. (folks use on non linux platforms)

Reply:
Yes, redirecting stderr is optional. We've not done it before either, with no complaints.

What about doing `gotestsum --raw-command <some script>` - is that worth pursuing?

Reply:
I don't have time for it, though - not until after the 1.31 release.